### DISCUSSION TITLE
zsh: permission denied: help

### DISCUSSION THREAD

**Discussion by DanKaplanSES** [microsoft/WSL#9436] [General]
### Version

Microsoft Windows [Version 10.0.19044.2364]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.4.72

### Distro Version

Ubuntu 20.04

### Other Software

zsh shell

### Repro Steps

For me, `help trap` outputs `zsh: permission denied: help`. None of these commands print anything:

```zsh
➜  ~ trap --help
➜  ~ trap --help
➜  ~ trap --h
➜  ~ trap -h
➜  ~
```

Simply typing `help` gives the same error. 

I don't have an ubuntu host OS to test on, so this error may occur universally. If so, please ignore.

NOTE: This is a duplicate of [#3833](https://github.com/microsoft/WSL/issues/3833), but I am creating this duplicate because the author closed theirs when they were given a workaround. I'm looking for a fix to the `zsh: permission denied: help` error.

### Expected Behavior

`help` in zsh prints the same output that a host Ubuntu OS would.

### Actual Behavior

This permission denied error is printed: `zsh: permission denied: help`

**Comment by elsaco** [comment]
`help` is not defined inside a zsh session unless explicitly done. `run-help` is a zsh function that provides the kind of help you tried above. It is part of `zsh-common` package that's picked up  automatically when installing zsh.

```
elsaco@RIPPER ~ % type help
help not found
elsaco@RIPPER ~ % type run-help
run-help is a shell function from /usr/share/zsh/functions/Misc/run-help
```

You could add an alias `help` if so desired, i.e. `alias help=run-help`.  Or just use `run-help`

**Comment by DanKaplanSES** [comment]
@elsaco Thanks for taking the time to help me. I should've clarified: I am aware of `run-help` and that `help` is for bash, not zsh. My concern is that `help` in zsh prints "permission denied." This doesn't occur for other nonexistent commands:

```bash
➜  ~ type asdf
asdf not found
➜  ~ asdf
zsh: command not found: asdf
```

Why doesn't `help` say `zsh: command not found: help`? Does the answer have anything to do with WSL? If no, then I will close this issue and maybe open a new one in the zsh repo. But--as I have no host machine with Ubuntu installed--I am unable to answer my own question\*. When I google this error message, WSL is mentioned in most of the results, which could indicate this is WSL specific and doesn't occur on host Ubuntu.  

\* Maybe I could use an Ubuntu docker container to test this and it would be the same as trying on an Ubuntu host machine? I guess hindsight is 20/20.

**Comment by elsaco** [comment]
IMO, it's not a WSL bug. What `zsh` version are you using? On `Ubuntu 22.0.4.1` and `zsh 5.8.1` it shows command not found:

```
dragon% zsh --version
zsh 5.8.1 (aarch64-unknown-linux-gnu)
dragon% type help
help not found
dragon% help
zsh: command not found: help
dragon% uname -r
5.15.79.1-microsoft-standard-WSL2
```

WSL used for testing:

```
PS C:\Users\elsaco> wsl --version
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.963

```

Update: after installing `Ubuntu 18.04.5` the permission denied error showed up:

```
dragon% zsh --version
zsh 5.4.2 (aarch64-unknown-linux-gnu)
dragon% type help
help not found
dragon% help
zsh: permission denied: help
```
